crates/namada/src/ledger/protocol/mod.rs

```diff
@@ -930,64 +936,47 @@ where
-                                InternalAddress::Multitoken,
-                            )))
+                            verifiers
+                                .contains(&Address::Internal(
+                                    InternalAddress::Multitoken,
+                                ))
+                                .ok_or_else(|| {
+                                    Error::AccessForbidden(
+                                        internal_addr.clone(),
+                                    )
+                                })
                         }
                         InternalAddress::Masp => {
                             let masp = MaspVp { ctx };
                             masp.validate_tx(tx, &keys_changed, &verifiers)
                                 .map_err(Error::MaspNativeVpError)
                         }
-                    };
-
-                    accepted.map_err(|err| {
-                        // No need to check invalid sig because internal vps
-                        // don't check the signature
-                        if sentinel.borrow().is_out_of_gas() {
-                            Error::GasError(err.to_string())
-                        } else {
-                            err
-                        }
-                    })
+                    }
                 }
             };
 
-            match accept {
-                Ok(accepted) => {
-                    if accepted {
-                        result.accepted_vps.insert(addr.clone());
-                    } else {
-                        result.rejected_vps.insert(addr.clone());
-                    }
-                }
-                Err(err) => match err {
-                    // Execution of VPs can (and must) be short-circuited
-                    // only in case of a gas overflow to prevent the
-                    // transaction from consuming resources that have not
-                    // been acquired in the corresponding wrapper tx. For
-                    // all the other errors we keep evaluating the vps. This
-                    // allows to display a consistent VpsResult across all
-                    // nodes and find any invalid signatures
-                    Error::GasError(_) => {
-                        return Err(err);
-                    }
-                    Error::InvalidTxSignature => {
-                        result.invalid_sig = true;
-                        result.rejected_vps.insert(addr.clone());
-                        // Don't push the error since this is just a flag error
-                    }
-                    _ => {
-                        result.rejected_vps.insert(addr.clone());
-                        result.errors.push((addr.clone(), err.to_string()));
-                    }
+            tx_accepted.map_or_else(
+                |err| {
+                    result.invalid_sig = err.is_invalid_section_signature(); // required by replay protection
```

Comment:
I believe we could remove this `invalid_sig` field from `VpsResult`. I think it was there before because previously an invalid signature would lead to an `Ok(false)` outcome but if everything is an error now we might just push the error to `VpsResult.errors` and then look for invalid signatures in `finalize_block`